src/main/java/com/suse/salt/netapi/calls/modules/Git.java

```diff
@@ -43,9 +44,32 @@ private Git() { }
 
     public static LocalCall<Boolean> branch(String cwd, String branch,
             String opts, String gitOpts, Optional<String> user) {
-        List<String> args = Arrays.asList(cwd, branch, opts, gitOpts);
+        List<String> args = new ArrayList<>(Arrays.asList(cwd, branch, opts, gitOpts));
```

Comment:
I wonder why this would be needed (here and above), wouldn't `Arrays.asList()` already create and return a list of strings as `List<String>`?

Reply:
Yes, you are right. It returns a list of strings as `List<String>`.  However the size of the list is fixed, this will not allow us to add other arguments to the list. Below the documentation of the method:
```

/**
     * Returns a fixed-size list backed by the specified array.  (Changes to
     * the returned list "write through" to the array.)  This method acts
     * as bridge between array-based and collection-based APIs, in
     * combination with {@link Collection#toArray}.  The returned list is
     * serializable and implements {@link RandomAccess}.
     *
     * <p>This method also provides a convenient way to create a fixed-size
     * list initialized to contain several elements:
     * <pre>
     *     List&lt;String&gt; stooges = Arrays.asList("Larry", "Moe", "Curly");
     * </pre>
     *
     * @param <T> the class of the objects in the array
     * @param a the array by which the list will be backed
     * @return a list view of the specified array
     */
    @SafeVarargs
    @SuppressWarnings("varargs")
    public static <T> List<T> asList(T... a) {
        return new ArrayList<>(a);
    }
```

If you add an element to that list, the application will throw an Exception. The current code actually send an exception, maybe it was not well tested. This is not the only line of code where this happens, so I made that change on in different places of the git module.

Reply:
Thanks, it makes sense!